xija/component/heat.py

```diff
@@ -903,3 +903,44 @@ def plot_data__time(self, fig, ax):
             ax.grid()
             ax.set_title('{}: data (blue)'.format(self.name))
             ax.set_ylabel('Power')
+
+class StepFunctionPower(PrecomputedHeatPower):
+    """
+    A class that applies a constant temperature shift only 
+    after a certain point in time.
+    """
+    def __init__(self, model, node, time, P=0.0):
+        super(StepFunctionPower, self).__init__(model)
+        self.time = DateTime(time).secs
+        self.node = self.model.get_comp(node)
+        self.add_par('P', P, min=-10.0, max=10.0)
+        self.n_mvals = 1
+
+    def __str__(self):
+        return 'step_power__{0}'.format(self.node)
+
+    def get_dvals_tlm(self):
+        """Return an array of zeros => no activation of the heater.
+        """
+        return np.zeros_like(self.model.times)
+
+    def update(self):
+        """Update the model prediction as a precomputed heat.
+        """
+        self.mvals = self.P*np.ones_like(self.model.times)
+        self.mvals[self.model.times < self.time] = 0.0
```

Comment:
Since `self.model.times` is sorted, it's going to be faster to use
```
idx0 = np.searchsorted(self.model.times, self.time)
self.mvals[:idx0] = 0.0
```
This should work for cases where `self.time` is either before or after the range of `model.times`, but worth checking.